### PR TITLE
#3145 Fix import of simple Task

### PR DESCRIPTION
**Pull request by CTI777**
1. During importing of xml process <task> tag will not be replaced with <manualTask> 
2. Incoming and outgoing flow elements are supported for task, userTask and manualTask

### CODE REVIEW

**Comment by CTI777** [comment]
Related to 
https://github.com/Activiti/Activiti/issues/3145

File: activiti-core/activiti-bpmn-converter/src/test/java/org/activiti/editor/language/xml/ManualTaskConverterTest.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    public void convertModelToXML() throws Exception {
+        BpmnModel bpmnModel = readXMLFile();
+        BpmnModel parsedModel = exportAndReadXMLFile(bpmnModel);
+        validateModel(parsedModel);
+        deployProcess(parsedModel);
+    }
+
+    @Override
+    protected String getResource() {
+        return "manualtaskmodel.bpmn";
```
Comment (erdemedeiros) [inline activiti-core/activiti-bpmn-converter/src/test/java/org/activiti/editor/language/xml/ManualTaskConverterTest.java]: Please, use the camel case in the file names.

File: activiti-core/activiti-bpmn-converter/src/test/java/org/activiti/editor/language/xml/SimpleUserTaskConverterTest.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    public void convertModelToXML() throws Exception {
+        BpmnModel bpmnModel = readXMLFile();
+        BpmnModel parsedModel = exportAndReadXMLFile(bpmnModel);
+        validateModel(parsedModel);
+        deployProcess(parsedModel);
+    }
+
+    @Override
+    protected String getResource() {
+        return "simpleusertaskmodel.bpmn";
```
Comment (erdemedeiros) [inline activiti-core/activiti-bpmn-converter/src/test/java/org/activiti/editor/language/xml/SimpleUserTaskConverterTest.java]: Camel case.

File: activiti-core/activiti-bpmn-converter/src/test/java/org/activiti/editor/language/xml/TaskConverterTest.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    public void convertModelToXML() throws Exception {
+        BpmnModel bpmnModel = readXMLFile();
+        BpmnModel parsedModel = exportAndReadXMLFile(bpmnModel);
+        validateModel(parsedModel);
+        deployProcess(parsedModel);
+    }
+
+    @Override
+    protected String getResource() {
+        return "taskmodel.bpmn";
```
Comment (erdemedeiros) [inline activiti-core/activiti-bpmn-converter/src/test/java/org/activiti/editor/language/xml/TaskConverterTest.java]: Camel case.

File: activiti-core/activiti-bpmn-converter/src/test/java/org/activiti/editor/language/xml/ManualTaskConverterTest.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        BpmnModel bpmnModel = readXMLFile();
+        validateModel(bpmnModel);
+    }
+
+    @Test
+    public void convertModelToXML() throws Exception {
+        BpmnModel bpmnModel = readXMLFile();
+        BpmnModel parsedModel = exportAndReadXMLFile(bpmnModel);
+        validateModel(parsedModel);
+        deployProcess(parsedModel);
```
Comment (erdemedeiros) [inline activiti-core/activiti-bpmn-converter/src/test/java/org/activiti/editor/language/xml/ManualTaskConverterTest.java]: This line can be removed. If you check the content method implementation, it's doing nothing.

File: activiti-core/activiti-bpmn-converter/src/test/java/org/activiti/editor/language/xml/SimpleUserTaskConverterTest.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        BpmnModel bpmnModel = readXMLFile();
+        validateModel(bpmnModel);
+    }
+
+    @Test
+    public void convertModelToXML() throws Exception {
+        BpmnModel bpmnModel = readXMLFile();
+        BpmnModel parsedModel = exportAndReadXMLFile(bpmnModel);
+        validateModel(parsedModel);
+        deployProcess(parsedModel);
```
Comment (erdemedeiros) [inline activiti-core/activiti-bpmn-converter/src/test/java/org/activiti/editor/language/xml/SimpleUserTaskConverterTest.java]: You can remove `deployProcess(parsedModel);`

File: activiti-core/activiti-bpmn-converter/src/test/java/org/activiti/editor/language/xml/SimpleUserTaskConverterTest.java [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    @Override
+    protected String getResource() {
+        return "simpleusertaskmodel.bpmn";
+    }
+
+    private void validateModel(BpmnModel model) throws Exception {
+        FlowElement flowElement = model.getMainProcess().getFlowElement("UserTask_0ej3luy");
+        assertNotNull(flowElement);
```
Comment (erdemedeiros) [inline activiti-core/activiti-bpmn-converter/src/test/java/org/activiti/editor/language/xml/SimpleUserTaskConverterTest.java]: Use AssertJ assertions.

File: activiti-core/activiti-bpmn-converter/src/test/java/org/activiti/editor/language/xml/TaskConverterTest.java [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+        BpmnModel bpmnModel = readXMLFile();
+        validateModel(bpmnModel);
+    }
+
+    @Test
+    public void convertModelToXML() throws Exception {
+        BpmnModel bpmnModel = readXMLFile();
+        BpmnModel parsedModel = exportAndReadXMLFile(bpmnModel);
+        validateModel(parsedModel);
+        deployProcess(parsedModel);
```
Comment (erdemedeiros) [inline activiti-core/activiti-bpmn-converter/src/test/java/org/activiti/editor/language/xml/TaskConverterTest.java]: You can remove `deployProcess(parsedModel);`

File: activiti-core/activiti-bpmn-converter/src/test/java/org/activiti/editor/language/xml/TaskConverterTest.java [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    @Override
+    protected String getResource() {
+        return "taskmodel.bpmn";
+    }
+
+    private void validateModel(BpmnModel model) throws Exception {
+        FlowElement flowElement = model.getMainProcess().getFlowElement("Task_17unmd8");
+        assertNotNull(flowElement);
```
Comment (erdemedeiros) [inline activiti-core/activiti-bpmn-converter/src/test/java/org/activiti/editor/language/xml/TaskConverterTest.java]: User AssertJ assertions.

File: activiti-core/activiti-bpmn-converter/src/test/resources/checkConversionPrefix.bpmn [hunk 9]
```diff
@@ -4,7 +4,10 @@
         <bpmn2:startEvent id="StartEvent_1">
             <bpmn2:outgoing>SequenceFlow_0afdugi</bpmn2:outgoing>
         </bpmn2:startEvent>
-        <bpmn2:manualTask id="Task_0cshed6" name="Task"></bpmn2:manualTask>
+        <bpmn2:manualTask id="Task_0cshed6" name="Task">
+        	<bpmn2:incoming>SequenceFlow_0afdugi</bpmn2:incoming>
```
Comment (erdemedeiros) [inline activiti-core/activiti-bpmn-converter/src/test/resources/checkConversionPrefix.bpmn]: The indentation contains tab characters. User spaces only.

File: activiti-core/activiti-bpmn-converter/src/test/java/org/activiti/editor/language/xml/ManualTaskConverterTest.java [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    @Override
+    protected String getResource() {
+        return "manualtaskmodel.bpmn";
+    }
+
+    private void validateModel(BpmnModel model) throws Exception {
+        FlowElement flowElement = model.getMainProcess().getFlowElement("ManualTask_0ej3luy");
+        assertNotNull(flowElement);
```
[the remark this answers is not in the document]
Comment (CTI777) [reply activiti-core/activiti-bpmn-converter/src/test/java/org/activiti/editor/language/xml/ManualTaskConverterTest.java]: fixed

File: activiti-core/activiti-bpmn-converter/src/test/java/org/activiti/editor/language/xml/ManualTaskConverterTest.java (hunk 1 above)
Comment (CTI777) [reply activiti-core/activiti-bpmn-converter/src/test/java/org/activiti/editor/language/xml/ManualTaskConverterTest.java]: fixed

File: activiti-core/activiti-bpmn-converter/src/test/java/org/activiti/editor/language/xml/SimpleUserTaskConverterTest.java (hunk 2 above)
Comment (CTI777) [reply activiti-core/activiti-bpmn-converter/src/test/java/org/activiti/editor/language/xml/SimpleUserTaskConverterTest.java]: fixed

File: activiti-core/activiti-bpmn-converter/src/test/java/org/activiti/editor/language/xml/ManualTaskConverterTest.java (hunk 4 above)
Comment (CTI777) [reply activiti-core/activiti-bpmn-converter/src/test/java/org/activiti/editor/language/xml/ManualTaskConverterTest.java]: removed

File: activiti-core/activiti-bpmn-converter/src/test/java/org/activiti/editor/language/xml/SimpleUserTaskConverterTest.java (hunk 5 above)
Comment (CTI777) [reply activiti-core/activiti-bpmn-converter/src/test/java/org/activiti/editor/language/xml/SimpleUserTaskConverterTest.java]: removed

File: activiti-core/activiti-bpmn-converter/src/test/java/org/activiti/editor/language/xml/SimpleUserTaskConverterTest.java (hunk 6 above)
Comment (CTI777) [reply activiti-core/activiti-bpmn-converter/src/test/java/org/activiti/editor/language/xml/SimpleUserTaskConverterTest.java]: fixed

File: activiti-core/activiti-bpmn-converter/src/test/java/org/activiti/editor/language/xml/TaskConverterTest.java (hunk 7 above)
Comment (CTI777) [reply activiti-core/activiti-bpmn-converter/src/test/java/org/activiti/editor/language/xml/TaskConverterTest.java]: removed

File: activiti-core/activiti-bpmn-converter/src/test/java/org/activiti/editor/language/xml/TaskConverterTest.java (hunk 8 above)
Comment (CTI777) [reply activiti-core/activiti-bpmn-converter/src/test/java/org/activiti/editor/language/xml/TaskConverterTest.java]: fixed

File: activiti-core/activiti-bpmn-converter/src/test/resources/checkConversionPrefix.bpmn (hunk 9 above)
Comment (CTI777) [reply activiti-core/activiti-bpmn-converter/src/test/resources/checkConversionPrefix.bpmn]: fixed

File: activiti-core/activiti-bpmn-converter/src/test/java/org/activiti/editor/language/xml/TaskConverterTest.java (hunk 3 above)
Comment (CTI777) [reply activiti-core/activiti-bpmn-converter/src/test/java/org/activiti/editor/language/xml/TaskConverterTest.java]: fixed

**Pull request merged**